packages/plugins/test/mocha/artifacts.spec.ts

```diff
@@ -93,30 +140,156 @@ describe('Artifacts', () => {
   it('makes artifacts for the third deployment on live networks', async () => {
     const firstArtifacts = await makeThenRunThenCheckDeployment(
       makeStandardDeployment(0, ExecutionMode.Platform, safeAddress),
-      {}
+      getEmptyDeploymentArtifacts()
     )
     const secondArtifacts = await makeThenRunThenCheckDeployment(
       makeStandardDeployment(1, ExecutionMode.Platform, safeAddress),
-      firstArtifacts.networks
+      firstArtifacts
     )
     await makeThenRunThenCheckDeployment(
       makeStandardDeployment(2, ExecutionMode.Platform, safeAddress),
-      secondArtifacts.networks
+      secondArtifacts
     )
   })
 
   it('makes artifacts for partially executed deployment', async () => {
     await makeThenRunThenCheckDeployment(
       makeRevertingDeployment(0, ExecutionMode.Platform, safeAddress),
-      {}
+      getEmptyDeploymentArtifacts()
     )
   })
 
   it('makes artifacts for remotely compiled deployment', async () => {
     await makeThenRunThenCheckRemoteDeployment(
       makeStandardDeployment(0, ExecutionMode.Platform, safeAddress),
-      {}
+      getEmptyDeploymentArtifacts()
+    )
+  })
+
+  // Tests a scenario where deployment artifacts are created on a chain when existing deployment
+  // artifacts already exist on a different chain. In this scenario, the deployment artifact
+  // logic must return the previous deployment artifacts in addition to the new ones.
+  it('keeps previous deployment artifacts', async () => {
+    const newChainId = '123'
+    const newCompilerInputId = 'newCompilerInputId'
+    const newContractName = 'MyContract2'
+    const newFullyQualifiedName = 'contracts/test/MyContracts.sol:MyContract2'
+    const newMerkleRoot = '0x' + '42'.repeat(32)
+
+    const deploymentConfig = await getFakeDeploymentConfig(
+      newChainId,
+      newFullyQualifiedName,
+      MyContract2Artifact.bytecode.object,
+      foundryToml.artifactFolder,
+      newCompilerInputId,
+      newMerkleRoot
+    )
+    const receipts = [getFakeActionSucceededReceipt(newMerkleRoot)]
+    const artifacts = getDummyDeploymentArtifacts()
+    const deployment = {
+      [newChainId]: {
+        deploymentConfig,
+        receipts,
+        provider: mockProvider,
+      },
+    }
+
+    await makeDeploymentArtifacts(
+      deployment,
+      newMerkleRoot,
+      deploymentConfig.configArtifacts,
+      artifacts
+    )
+
+    const previousContractArtifact =
+      artifacts.networks[dummyChainId].contractDeploymentArtifacts[
+        `${dummyContractName}.json`
+      ]
+    const newContractArtifact =
+      artifacts.networks[newChainId].contractDeploymentArtifacts[
+        `${newContractName}.json`
+      ]
+    expect(isContractDeploymentArtifact(previousContractArtifact)).equals(true)
+    expect(previousContractArtifact.chainId).equals(dummyChainId)
+    expect(isContractDeploymentArtifact(newContractArtifact)).equals(true)
+    expect(newContractArtifact.chainId).equals(newChainId)
```

Comment:
Agree that it would be better, but I'm gonna keep this unchanged because this test case checks for different things than the `checkArtifacts` function, and it'd be nontrivial to update the `checkArtifacts` function. Specifically, the `checkArtifacts` function assumes that the networks and contracts remain the same for all deployments, whereas this test case changes the network and the contract in each deployment.